homeassistant/components/mqtt/mixins.py

```diff
@@ -337,6 +339,117 @@ async def _async_setup_entities() -> None:
     await _async_setup_entities()
 
 
+async def async_mqtt_entry_helper(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    entity_class: type[MqttEntity],
+    domain: str,
+    async_add_entities: AddEntitiesCallback,
+    discovery_schema: vol.Schema,
+    platform_schema_modern: vol.Schema,
+) -> None:
+    """Set up entity, automation or tag creation dynamically through MQTT discovery."""
+    mqtt_data = get_mqtt_data(hass)
+
+    async def async_discover(discovery_payload: MQTTDiscoveryPayload) -> None:
```

Comment:
If we don't need to await inside we can make this a callback.

Reply:
Same here: might work for entity platforms but not for the entity-platforms.

Reply:
Ok